### PR TITLE
Add longer descriptions to `blueprints` impls

### PR DESCRIPTION
**Pull request by keckler**
## What is the change?

Just adding long-form descriptions to the impl tags in the `blueprints` package.

## Why is the change being made?

Quality

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
I believe @kasticrunch volunteered to review this PR.

File: armi/reactor/blueprints/assemblyBlueprint.py [hunk 1]
```diff
@@ -100,6 +118,23 @@ class AssemblyBlueprint(yamlize.Object):
...
         :id: I_ARMI_BP_ASSEM
         :implements: R_ARMI_BP_ASSEM
+
+        Defines a yaml construct that allows the user to specify attributes of an
+        assembly from within their blueprints file, including a name, flags, specifier
+        for use in defining a core map, a list of blocks, a list of block heights,
+        a list of axial mesh points in each block, a list of cross section identifiers
+        for each block, and material options (see :need:`I_ARMI_MAT_USER_INPUT0`).
+
+        Relies on the underlying infrastrature from the ``yamlize`` package for
```
Comment (kasticrunch) [inline armi/reactor/blueprints/assemblyBlueprint.py:128]:
```suggestion
        Relies on the underlying infrastructure from the ``yamlize`` package for
```

File: armi/reactor/blueprints/blockBlueprint.py [hunk 2]
```diff
@@ -47,6 +47,26 @@ class BlockBlueprint(yamlize.KeyedList):
...
+        Defines a yaml construct that allows the user to specify attributes of a
+        block from within their blueprints file, including a name, flags, a radial
+        grid to specify locations of pins, and the name of a component which
+        drives the axial expansion of the block (see :py:mod:`~armi.reactor.converters.axialExpansionChanger`).
+
+        In addition, the user may specify key-value pairs to specify the components
+        contained within the block, where the keys are component names and the
+        values are component blueprints (see :py:class:`~armi.reactor.blueprints.ComponentBlueprint.ComponentBlueprint`).
+
+        Relies on the underlying infrastrature from the ``yamlize`` package for
```
Comment (kasticrunch) [inline armi/reactor/blueprints/blockBlueprint.py:60]:
```suggestion
        Relies on the underlying infrastructure from the ``yamlize`` package for
```

File: armi/reactor/blueprints/componentBlueprint.py [hunk 3]
```diff
@@ -123,6 +123,30 @@ class ComponentBlueprint(yamlize.Object):
...
+        material and/or isotopic vector, input temperature, corresponding component dimensions,
+        and ID for placement in a block lattice (see :py:class:`~armi.reactor.blueprints.blockBlueprint.BlockBlueprint`).
+        Component dimensions that can be defined for a given component are dependent
+        on the component's ``shape`` attribute, and the dimensions defining each
+        shape can be found in the :py:mod:`~armi.reactor.components` module.
+
+        Limited validation on the inputs is performed to ensure that the component
+        shape corresponds to a valid shape defined by the ARMI application.
+
+        Relies on the underlying infrastrature from the ``yamlize`` package for
```
Comment (kasticrunch) [inline armi/reactor/blueprints/componentBlueprint.py:138]:
```suggestion
        Relies on the underlying infrastructure from the ``yamlize`` package for
```

File: armi/reactor/blueprints/gridBlueprint.py [hunk 4]
```diff
@@ -148,6 +148,24 @@ class GridBlueprint(yamlize.Object):
         :id: I_ARMI_BP_GRID
         :implements: R_ARMI_BP_GRID
 
+        Defines a yaml construct that allows the user to specify a grid
+        from within their blueprints file, including a name, geometry, dimensions,
+        symmetry, and a map with the relative locations of components within that grid.
+
+        Relies on the underlying infrastrature from the ``yamlize`` package for
```
Comment (kasticrunch) [inline armi/reactor/blueprints/gridBlueprint.py:155]:
```suggestion
        Relies on the underlying infrastructure from the ``yamlize`` package for
```

File: armi/reactor/blueprints/isotopicOptions.py [hunk 5]
```diff
@@ -154,6 +172,26 @@ class CustomIsotopic(yamlize.Map):
     .. impl:: Certain material modifications will be applied using this code.
         :id: I_ARMI_MAT_USER_INPUT2
         :implements: R_ARMI_MAT_USER_INPUT
+
+        Defines a yaml construct that allows the user to define a custom isotopic vector
+        from within their blueprints file, including a name and key-value pairs
+        corresponding to nuclide names and their concentrations.
+
+        Relies on the underlying infrastrature from the ``yamlize`` package for
```
Comment (kasticrunch) [inline armi/reactor/blueprints/isotopicOptions.py:180]:
```suggestion
        Relies on the underlying infrastructure from the ``yamlize`` package for
```

**Review by kasticrunch — CHANGES_REQUESTED** [review]

**Review by kasticrunch — APPROVED** [review]

**Comment by kasticrunch** [comment]
Reviewed and approving the added documentation to the implementation tags and other editorial changes.

**Pull request merged**